File: app/src/main/java/com/example/myapplication/AlertFragment.java
package com.example.myapplication;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Build;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.RequiresApi;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;

import java.util.Optional;


public class AlertFragment extends DialogFragment {

    private String mMsg;

    public AlertFragment(String msg){
        mMsg = Optional.ofNullable(msg).orElse("wrong credentials");
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        builder.setTitle("Error!")
                .setMessage(mMsg)
                .setIcon(R.drawable.ic_baseline_error)
                .setPositiveButton("ОК", new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int id) {
                        dialog.cancel();
                    }
                });
        return builder.create();
    }
}

File: app/src/main/java/com/example/myapplication/AlertFragment.kt
package com.example.myapplication

import android.app.Dialog
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment

class AlertFragment(msg: String?) : DialogFragment() {
    private val mMsg: String = msg ?:"wrong credentials"
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(
            requireActivity()
        )
        builder.setTitle("Error!")
            .setMessage(mMsg)
            .setIcon(R.drawable.ic_baseline_error)
            .setPositiveButton("ОК") { dialog, _ -> dialog.cancel() }
        return builder.create()
    }
}